bundles/org.openhab.core.io.websocket/src/main/java/org/openhab/core/io/websocket/CommonWebSocketServlet.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.io.websocket;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import javax.ws.rs.core.SecurityContext;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.websocket.server.WebSocketServerFactory;
+import org.eclipse.jetty.websocket.servlet.ServletUpgradeRequest;
+import org.eclipse.jetty.websocket.servlet.ServletUpgradeResponse;
+import org.eclipse.jetty.websocket.servlet.WebSocketCreator;
+import org.eclipse.jetty.websocket.servlet.WebSocketServlet;
+import org.eclipse.jetty.websocket.servlet.WebSocketServletFactory;
+import org.openhab.core.auth.AuthenticationException;
+import org.openhab.core.auth.Role;
+import org.openhab.core.io.rest.auth.AnonymousUserSecurityContext;
+import org.openhab.core.io.rest.auth.AuthFilter;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.http.NamespaceException;
+import org.osgi.service.http.whiteboard.propertytypes.HttpWhiteboardServletName;
+import org.osgi.service.http.whiteboard.propertytypes.HttpWhiteboardServletPattern;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link CommonWebSocketServlet} provides the servlet for WebSocket connections
+ *
+ * @author Jan N. Klug - Initial contribution
+ * @author Miguel Álvarez Díez - Refactor into a common servlet
+ */
+@NonNullByDefault
+@HttpWhiteboardServletName(CommonWebSocketServlet.SERVLET_PATH)
+@HttpWhiteboardServletPattern(CommonWebSocketServlet.SERVLET_PATH + "/*")
+@Component(immediate = true, service = { Servlet.class })
+public class CommonWebSocketServlet extends WebSocketServlet {
+    private static final long serialVersionUID = 1L;
+
+    public static final String SERVLET_PATH = "/ws";
+
+    public static final String DEFAULT_HANDLER_ID = EventWebSocketHandler.HANDLER_ID;
+
+    private final Map<String, WebSocketHandler> connectionHandlers = new HashMap<>();
+    private final AuthFilter authFilter;
+
+    @SuppressWarnings("unused")
+    private @Nullable WebSocketServerFactory importNeeded;
+
+    @Activate
+    public CommonWebSocketServlet(@Reference AuthFilter authFilter) throws ServletException, NamespaceException {
```

Comment:
Use `org.openhab.core.auth.AuthenticationManager` instead of filter, filter itself is an entry point to authentication process, not its driver.

Reply:
The reason to use use the AuthFilter over the AuthenticationManager is to make it behave in the same way than the rest api and respect its configuration (allowBasicAuth, implicitUserRole and trustedNetworks).
Do you think is it better to refactor that functionality into a new class and make use of it in both places?

Reply:
Its a design question - you can think of `trustedNetwork` as another level of credentials (caller IP address), which has nothing to do with username/password. Basic auth is matter of extracting the credentials from headers, thus it is not an authentication provisioning mechanism, it is credential source. For implicit user in role - I am not sure what's purpose of that thing is, hence I cant comment on it.

Reply:
The "implicitUserRole" option makes the token not required and grant any request without token with user credentials (gives an AnonymousUserContext).

The "trustedNetworks" option allows to disable the "implicitUserRole" but to preserve this behavior in for some networks filtering the request ip by CIDR.

Yes both three options are not authentication provisioning mechanisms but affect to whetter this runs.

Another thing to highlight, the JWT token is only handled in this bundle so not supported by the org.openhab.core.auth.AuthenticationManager.

Reply:
JWT token can be handled through `AuthenticationProvider` too, see:

JWTCredentials:
https://github.com/opensmarthouse/opensmarthouse-core/commit/947e116347df8f813322319d8224ebb698f167fe#diff-cdb1dc31859fa2b30a4312f8a9c5faba7b2594a311cffae0a4fcb7c3dabbfa09

JwtAuthenticationProvider (an external JWT)
https://github.com/opensmarthouse/opensmarthouse-core/commit/947e116347df8f813322319d8224ebb698f167fe#diff-30f808a93ce7a81e41fa6d0335d763c3d8f319577854db8ad5a9eb3523eaf6a4

LocalJwtAuthenticationProvider:
https://github.com/opensmarthouse/opensmarthouse-core/commit/947e116347df8f813322319d8224ebb698f167fe#diff-8100f63b9cbf2eb2971e7f074f9054c6c2691b0a114920ab0218d92a562e64e0

Anyhow, these are samples I made in a fork which do not apply here. Also your PR is not about revisiting security. Lets focus on not making too many assumptions on top of `AuthFilter` so eventual re-organization of this layer after 4.0 release will not result in a complete rewrite of logic you supply.

Reply:
Sure, thank you for the samples anyway.

Reply:
With the changes done in the AuthFilter it should be easier to refactor the code, just need the 'getSecuryContext' functionality to be provided from the core. But I think that is better to let that for a future refactor as you said.